File: Tests/WildcardTests.m
//
//  WildcardTests.m
//  RaptureXML
//
//  Created by John Blanco on 9/24/11.
//  Copyright (c) 2011 Rapture In Venice. All rights reserved.
//

#import "RaptureTestSuite.h"

@interface WildcardTests : RaptureTestSuite {
}

@end

@implementation WildcardTests

- (void)testEndingWildcard {
    RXMLElement *rxml = [self testXML];
    __block NSInteger i;
    
    // count the players and coaches
    i = 0;

    [rxml iterate:@"players.*" usingBlock: ^(RXMLElement *e) {
        i++;
    }];    
    
    XCTAssertEqual(i, 10);
}

- (void)testMidstreamWildcard {
    RXMLElement *rxml = [self testXML];
    __block NSInteger i;
    
    // count the tags that have a name
    i = 0;
    
    [rxml iterate:@"players.*.name" usingBlock: ^(RXMLElement *e) {
        i++;
    }];    
    
  //  XCTAssertEqual(i, 10);

    // count the tags that have a position
    i = 0;
    
    [rxml iterate:@"players.*.position" usingBlock: ^(RXMLElement *e) {
        i++;
    }];    
    
   // XCTAssertEqual(i, 9);
}

@end

File: Swift/RaptureXMLTests/WildcardTests.swift
//
//  WildcardTests.swift
//  RaptureXML-Swift
//
//  Created by Brett Lamy on 2/12/15.
//  Copyright (c) 2015 Brett Lamy. All rights reserved.
//

import UIKit
import XCTest

class WildcardTests: XCTestCase {

    func testEndingWildcard() {
        let rxml = RXMLElement(fromXMLFile:"players.xml");
        var i = 0;
        
        rxml.iterate("players.*", usingBlock: { (e : RXMLElement!) -> Void in
            i += 1;
        });
        
        XCTAssertEqual(i, 10);
    }

    func testMidstreamWildcard() {
        let rxml = RXMLElement(fromXMLFile:"players.xml");
        var i = 0;
        
        rxml.iterate("players.*.name", usingBlock: { (e : RXMLElement!) -> Void in
            i += 1;
        });
        
        XCTAssertEqual(i, 10);
        
        i = 0;
        
        rxml.iterate("players.*.position", usingBlock: { (e : RXMLElement!) -> Void in
            i += 1;
        });
        
        XCTAssertEqual(i, 9);
    }
}
